### PR TITLE
Added a cache based on mtime for watch/serve mode in esbuild

### PR DESCRIPTION
**Pull request by STRd6**
This speeds up esbuild serve quite a bit on larger projects.

### CODE REVIEW

File: integration/unplugin/src/index.ts [hunk 1]
```diff
@@ -363,7 +367,13 @@ export const rawPlugin: Parameters<typeof createUnplugin<PluginOptions>>[0] =
       const basename = id.slice(0, match.index + match[1].length);
 
       const filename = path.resolve(rootDir, basename);
-      const rawCivetSource = await fs.promises.readFile(filename, 'utf-8');
+      const mtime = fs.statSync(filename).mtimeMs;
```
Comment (edemaine) [inline integration/unplugin/src/index.ts:370]: Perhaps this should be a avoided if `cache` is null?

File: integration/unplugin/src/index.ts [hunk 2]
```diff
@@ -363,7 +367,13 @@ export const rawPlugin: Parameters<typeof createUnplugin<PluginOptions>>[0] =
...
+      const cached = cache?.get(filename);
+      if (cached && cached.mtime === mtime) {
+        return cached.result;
+      }
+
+      const rawCivetSource = fs.readFileSync(filename, 'utf-8');
```
Comment (edemaine) [inline integration/unplugin/src/index.ts:376]: Is there a reason to switch to sync interface? I wonder if promises interface (also stat) might support multitasking slightly better.

**Review by edemaine — APPROVED** [review]
Nice idea! Can you also document this? I think in a couple of READMEs.

File: integration/unplugin/src/index.ts (hunk 2 above)
Comment (STRd6) [reply integration/unplugin/src/index.ts:376]: My intuition is that opening a lot of files and reading a little bit of each is slower than opening one and reading all of it but I haven't done a full benchmark yet.

Comment (edemaine) [reply integration/unplugin/src/index.ts]: Ah, fair point - I'm not sure what's best. Probably fine either way.

**Review by edemaine — APPROVED** [review]
Looks good! I think `integrations/unplugin/README.md` also has a section.

File: integration/unplugin/src/index.ts (hunk 2 above)
Comment (STRd6) [reply integration/unplugin/src/index.ts]: I did a brief unscientific test and the promises version seemed to finish faster. The promises version probably has better throughput where node/the os can buffer multiple files in parallel.

**Pull request merged**